common/src/main/java/com/microsoft/identity/common/internal/broker/ipc/AccountManagerAddAccountStrategy.java

```diff
@@ -0,0 +1,103 @@
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.exception.BrokerCommunicationException;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.util.ProcessUtil;
+
+import java.io.IOException;
+
+/**
+ * A strategy for communicating with the targeted broker via AccountManager's addAccount().
+ *
+ * NOTE: SuppressLint is added because this API requires MANAGE_ACCOUNTS for API<= 22.
+ * AccountManagerUtil.canUseAccountManagerOperation() will validate that.
+ * */
+
+@SuppressLint("MissingPermission")
+public class AccountManagerAddAccountStrategy implements IIpcStrategy {
+    private static final String TAG = AccountManagerAddAccountStrategy.class.getSimpleName();
+
+    private final Context mContext;
+
+    public AccountManagerAddAccountStrategy(final Context context) {
+        mContext = context;
+    }
+
+    @Override
+    @Nullable public Bundle communicateToBroker(@NonNull final BrokerOperationBundle brokerOperationBundle)
+            throws BaseException {
+        final String methodName = brokerOperationBundle.getOperation().name();
+        try {
+            final AccountManagerFuture<Bundle> resultBundle =
+                    AccountManager.get(mContext)
+                            .addAccount(
+                                    AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE,
+                                    AuthenticationConstants.Broker.AUTHTOKEN_TYPE,
+                                    null,
+                                    getAccountManagerBundle(brokerOperationBundle),
+                                    null,
+                                    null,
+                                    ProcessUtil.getPreferredHandler());
+
+            Logger.verbose(TAG + methodName, "Received result from broker");
+            return resultBundle.getResult();
+        } catch (final AuthenticatorException | IOException | OperationCanceledException e) {
+            Logger.error(TAG + methodName, e.getMessage(), e);
+            throw new BrokerCommunicationException("Failed to connect to AccountManager", e);
+        }
+    }
+
+    /**
+     * Packs the response bundle with the account manager key.
+     */
+    public Bundle getAccountManagerBundle(@NonNull final BrokerOperationBundle brokerOperationBundle)
+            throws BrokerCommunicationException {
+        Bundle requestBundle = brokerOperationBundle.getBundle();
+        if (requestBundle == null) {
+            requestBundle = new Bundle();
+        }
+
+        requestBundle.putString(
+                AuthenticationConstants.Broker.BROKER_ACCOUNT_MANAGER_OPERATION_KEY,
+                brokerOperationBundle.getAccountManagerAddAccountOperationKey());
+
+        return requestBundle;
```

Comment:
Should we put this into a method on BrokerOperationBundle instead of putting it here?  Maybe putRequestString(key, value) that could contain all of this logic (check if the request bundle is created, create if necessary, add appropriate string under the correct specified key)?